src/main/scala/JsonSchemaParser.scala

```diff
@@ -234,11 +245,16 @@ object JsonSchemaParser {
   private def parseSchemaOpt(obj: ujson.Obj, elemName: String) =
     runOptParser(obj, elemName, optParser(parseSchema))
 
-  private def parseSchema(value: ujson.Value, errMsg: String) =
-    for {
-      obj <- value.objOpt.toRight(ParserError(errMsg))
-      schema <- parseSubSchema(obj)
-    } yield schema
+  private def parseSchema(value: ujson.Value, errMsg: String) = {
+    value match {
+      case ujson.Bool(v) => Right(Left(v))
+      case ujson.Obj(obj) =>
+        for {
+          schema <- parseSubSchema(obj)
+        } yield schema
+      case _ => Left(ParserError(errMsg))
```

Comment:
"Expected Bool or Obj: $errMsg" or something like that would be a better error message